File: 2_num_sum/2_num_sum.py
def get_2_num_sum(arr, target):
    buf = {}
    for num in arr:
        if num in buf:
            return [num, buf[num]]
        else:
            buf[target - num] = num
    return []


if __name__ == '__main__':
    print(get_2_num_sum([2, 9, -1, 3, 10, 11], 8))

File: 2_num_sum/2_num_sum.go
package main

import "fmt"

func twoNumSum(arr *[]int, target int) []int {
	buff := make(map[int]int)

	for _, num := range *arr {
		if value, found := buff[num]; found {
			return []int{value, num}
		}
		buff[target-num] = num
	}
	return []int{}
}

func main() {
	fmt.Println(twoNumSum(&[]int{2, 9, -1, 3, 10, 11}, 8))
}
